php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsCheck.java

```diff
@@ -144,9 +147,20 @@ private void checkForCredentialQuery(LiteralTree literal) {
   }
 
   private void checkForCredentialUri(LiteralTree literal) {
-    Matcher m = URI_PATTERN.matcher(literal.value());
-    if (m.find() && !m.group(2).equals(m.group(3))) {
-      context().newIssue(this, literal, MESSAGE_URI);
+    String possibleUrl = CheckUtils.trimQuotes(literal.value());
+    URI uri = null;
+
+    try {
+      uri = new URI(possibleUrl);
+    } catch (URISyntaxException e) {
+      return;
+    }
+
+    if (uri.getUserInfo() != null) {
+      Matcher m = Pattern.compile("(\\S+):(\\S+)").matcher(uri.getUserInfo());
+      if (m.find() && !m.group(1).equals(m.group(2)) && !DEFAULT_CREDENTIAL_URI_PATTERN.matcher(uri.getUserInfo()).find()) {
```

Comment:
Rather than using a regular expression for `DEFAULT_CREDENTIAL_URI_PATTERN`, I believe it would be simpler to have something like `userInfo.equals("user:password") || userInfo.equals("username:password")`.

Reply:
Removed